### DISCUSSION TITLE
CrossSectionByPlane

### DISCUSSION THREAD

**Discussion by cartesian-theatrics** [elalish/manifold#498] [Ideas]
Hello,

I have an application where I'd like to be able take a CrossSection slice of a Manifold, something like Manifold.CrossSectionByPlane() -> CrossSection.

(1) Do you think this is a heavy lift?
(2) Could it be very high performance?

I'm considering trying to take a crack at it.

**Comment by geoffder** [comment]
I've only gave it a light reading, but this [implementation](https://github.com/intents-software/mesh-plane-intersection) I just looked up seems to be pretty straight forward (not so different to how I naively think about the problem), and is only a few hundred lines. How optimal it is, I don't know.

**Comment by geoffder** [comment]
Also I should mention, there is an issue for this already: https://github.com/elalish/manifold/issues/426.

**Reply by elalish** [reply]
Indeed, I would definitely like to have this function! And no, it shouldn't be too hard. For API, I would say we should only cut on the XY plane (like OpenSCAD), as a user can apply a rotation & translation to get there (we may want a helper method). Then say every edge with one `vert.z > 0` and the other `vert.z <= 0` creates one polygon point. I would probably just find all the edges that cross, then pick one and follow the halfedge neighbors around the manifold to create the polygon. Mark each edge as used, then pick a new unused one and repeat every time you close a loop. That should be O(n), which I think is about the best you can do.
